src/java.desktop/share/classes/sun/java2d/cmm/lcms/LCMS.java

```diff
@@ -106,10 +64,27 @@ public synchronized void setTagData(Profile p, int tagSignature, byte[] data) {
      * Throws CMMException if operation fails, preserve old profile from
      * destruction.
      */
-    private native void setTagDataNative(long ptr, int tagSignature,
-                                               byte[] data);
+    static native void setTagDataNative(long ptr, int tagSignature, byte[] data);
+    static native byte[] getProfileDataNative(long ptr);
+    static native byte[] getTagNative(long profileID, int signature);
+    private static native long loadProfileNative(byte[] data, Object ref);
+
+    @Override
+    public byte[] getProfileData(Profile p) {
+        return getLcmsProfile(p).getProfileData();
+    }
+
+    @Override
+    public byte[] getTagData(Profile p, int tagSignature) {
+        return getLcmsProfile(p).getTag(tagSignature);
+    }
+
+    @Override
+    public synchronized void setTagData(Profile p, int tagSignature, byte[] data) {
+        getLcmsProfile(p).setTag(tagSignature, data);
```

Comment:
Note that this method is synchronized, I think it is not needed to be, because a long time ago(before the LCMSProfile class was implemented ) all method in this class was synchronized to guard information about tags. But later we changed all such synchronization methods to the per-profile locks. And only this method remains synchronized.

But I leave it as is for now, since it might affect transforms which I am not touching in this fix. WIll remove that later.